internal/cmd/catalog/get.go

```diff
@@ -0,0 +1,131 @@
+	"github.com/mia-platform/miactl/internal/util"
+	"github.com/spf13/cobra"
+)
+
+const (
+	getItemByItemIDAndVersionEndpointTemplate = "/api/tenants/%s/marketplace/items/%s/versions/%s"
+
+	cmdGetLongDescription = `Get a single Catalog item
+
+	This command works with Mia-Platform Console v14.0.0 or later.
+
+	You need to specify the companyId, itemId and version, via the respective flags. The company-id flag can be omitted if it is already set in the context.
+	`
+	cmdGetUse = "get { --item-id item-id --version version }"
+)
+
+// GetCmd return a new cobra command for getting a single catalog resource
+func GetCmd(options *clioptions.CLIOptions) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   cmdGetUse,
+		Short: "Get Catalog item",
+		Long:  cmdGetLongDescription,
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			restConfig, err := options.ToRESTConfig()
+			cobra.CheckErr(err)
+			client, err := client.APIClientForConfig(restConfig)
+			cobra.CheckErr(err)
+
+			canUseNewAPI, versionError := util.VersionCheck(cmd.Context(), client, 14, 0)
+			if !canUseNewAPI || versionError != nil {
+				return catalog.ErrUnsupportedCompanyVersion
+			}
+
+			serializedItem, err := getItemEncodedWithFormat(
+				cmd.Context(),
+				client,
+				restConfig.CompanyID,
+				options.MarketplaceItemID,
+				options.MarketplaceItemVersion,
+				options.OutputFormat,
+			)
+			cobra.CheckErr(err)
+
+			fmt.Println(serializedItem)
+			return nil
+		},
+	}
+
+	options.AddOutputFormatFlag(cmd.Flags(), encoding.JSON)
+
+	itemIDFlagName := options.AddMarketplaceItemIDFlag(cmd.Flags())
+	versionFlagName := options.AddMarketplaceVersionFlag(cmd.Flags())
+
+	cmd.MarkFlagsRequiredTogether(itemIDFlagName, versionFlagName)
+
+	return cmd
+}
+
+func getItemByItemIDAndVersion(ctx context.Context, client *client.APIClient, companyID, itemID, version string) (*catalog.Item, error) {
+	endpoint := fmt.Sprintf(getItemByItemIDAndVersionEndpointTemplate, companyID, itemID, version)
+	return performGetItemRequest(ctx, client, endpoint)
+}
+
+func performGetItemRequest(ctx context.Context, client *client.APIClient, endpoint string) (*catalog.Item, error) {
+	resp, err := client.Get().APIPath(endpoint).Do(ctx)
+
+	if err != nil {
+		return nil, fmt.Errorf("error executing request: %w", err)
+	}
+
+	if err := resp.Error(); err != nil {
+		return nil, err
+	}
+
+	var marketplaceItem *catalog.Item
+	if err := resp.ParseResponse(&marketplaceItem); err != nil {
+		return nil, fmt.Errorf("error parsing response body: %w", err)
+	}
+
+	if marketplaceItem == nil {
+		return nil, fmt.Errorf("no catalog item returned in the response")
+	}
+
+	return marketplaceItem, nil
+}
+
+// getItemEncodedWithFormat retrieves the catalog item corresponding to the specified identifier, serialized with the specified outputFormat
+func getItemEncodedWithFormat(ctx context.Context, client *client.APIClient, companyID, itemID, version, outputFormat string) (string, error) {
+	var item *catalog.Item
```

Comment:
tip for a more idiomatic Go: these two definition can be omitted if you use `item, err := ...` at line 119 :)